tools/c7n_azure/c7n_azure/actions/lock.py

```diff
@@ -77,6 +102,15 @@ def _process_resource(self, resource):
                 ResourceIdParser.get_resource_name(resource.get('c7n:parent-id')) or '',
                 ResourceIdParser.get_resource_type(resource['id']),
                 resource['name'],
-                'custodian_lock_' + resource['name'] + '_' + self.lock_type,
-                ManagementLockObject(level=self.lock_type)
+                lock_name,
+                ManagementLockObject(level=self.lock_type, notes=lock_notes)
             )
+
+    def _get_lock_name(self, resource):
+        return self.data.get('lock-name',
+                             "custodian_lock_{}_{}".format(resource['name'], self.lock_type))
```

Comment:
What is the length limit for the lock name?

Reply:
"The name of lock. The lock name can be a maximum of 260 characters. It cannot contain <, > %, &, :, , ?, /, or any control characters."

Figured that if they input an invalid name, the API would return this info

Reply:
Ya I think I prefer we don't encode this verification, because it super easy to get out of date with it.

Reply:
I don't think you need to go crazy with regex validation but for similar behavior with NSG rules I believe we did at least set the JSON Schema `"minLength": x, "maxLength": x` properties for some basic sanity checks.

Reply:
Added min and max